lib/manager/focus_manager.dart

```diff
@@ -79,18 +67,20 @@ class FocusManager implements InputHandler {
   /// handling input. Components should be re-registered every frame after
   /// layout, since [componentInstances] is cleared on [reset].
   void register(InteractableComponentInstance c) {
-    componentInstances.add(c);
+    context.componentInstances.add(c);
```

Comment:
Can you add a check here that if context.currentComponent is not null then it should throw a exception/warning.